# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmDebug.cpp
# Commit: 8c7ce56c181b
# Full Hash: 8c7ce56c181bb521dd74a9754fcac855fc26fe32
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2022-03-29 09:56:04
# Description:
#   Bug 1761606 - wasm: Clean up debug filter allocation. r=lth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D142157
# ==============================================================================

diff -r 09e7d05bc460 -r 8c7ce56c181b js/src/wasm/WasmDebug.cpp
--- a/js/src/wasm/WasmDebug.cpp	Mon Mar 28 22:01:09 2022 +0000
+++ b/js/src/wasm/WasmDebug.cpp	Mon Mar 28 22:05:03 2022 +0000
@@ -23,12 +23,12 @@
 #include "debugger/Debugger.h"
 #include "ds/Sort.h"
 #include "jit/MacroAssembler.h"
-#include "wasm/WasmInstance.h"
 #include "wasm/WasmJS.h"
 #include "wasm/WasmStubs.h"
 #include "wasm/WasmValidate.h"
 
 #include "gc/GCContext-inl.h"
+#include "wasm/WasmInstance-inl.h"
 
 using namespace js;
 using namespace js::jit;
@@ -316,6 +316,7 @@
     return;
   }
 
+  MOZ_RELEASE_ASSERT(&instance->metadata() == &metadata());
   uint32_t numFuncs = metadata().debugFuncReturnTypes.length();
   if (enabled) {
     MOZ_ASSERT(enterAndLeaveFrameTrapsCounter_ > 0);